detox/test/ios/example/NativeModule.m

```diff
@@ -87,4 +87,19 @@ @implementation NativeModule
 	});
 }
 
+RCT_EXPORT_METHOD(presentOverlayWindow) {
+    static UIWindow *overlayWindow;
+
+    dispatch_async(dispatch_get_main_queue(), ^{
+        CGRect screenBounds = UIScreen.mainScreen.bounds;
+        overlayWindow = [[UIWindow alloc] initWithFrame:screenBounds];
+
+        [overlayWindow setWindowLevel:UIWindowLevelStatusBar];
+        [overlayWindow setBackgroundColor:[UIColor.grayColor colorWithAlphaComponent:0.5]];
```

Comment:
Would it make sense to challenge further and use a 0 alpha?

Reply:
Holy s**t, that's an excellent question!
If the overlying view has a transparent background (or, `clearColor`), we consider the underlying view as visible. So in this case the views are considered as "visible" but they are not tappable / scrollable. 

So it makes a lot of sense, but the tests will break, but we need to cover these cases also since they shouldn't (touch gestures should not pass).

We must have some better mechanism here than just asserting on visibility.